### DISCUSSION TITLE
Keyed Services in the IServiceProvider in .NET 8 preview 7

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#79] [General]
# Keyed Services in the IServiceProvider in .NET 8 preview 7

The .NET 8 preview 7 will bring another exciting feature some of you probably awaiting for a long time: Keyed services.

https://steven-giesel.com/blogPost/40dab3bb-d17e-4357-b579-b1d95fb9bfda

**Comment by tranb3r** [comment]
Actually, `IKeyedServiceProvider` does not exist, you have to use `IServiceProvider`.

```
class SmallCacheConsumer(IServiceProvider serviceProvider)
{
    public object? GetData() => serviceProvider.GetRequiredKeyedService<IMemoryCache>("big");
}
```

**Reply by linkdotnet** [reply]
Thanks. Yes, I took the original example from David Fowler. The API changed. I will adopt the article👍
